client/lib/user-settings/index.js

```diff
@@ -54,6 +54,23 @@ function deleteUnsavedSetting( settings, settingName, recursive ) {
 	}
 }
 
+/*
+ * Checks if an incoming change to settings.language is a change to the existing settings
+ * Currently the assumption is that if a settings.locale_variant slug exists, then that is the current language
+ */
+function hasLanguageChanged( languageSettingValue, settings = {} ) {
+	if ( ! languageSettingValue ) {
+		false;
+	}
+	// if there is a saved variant we know that the user is changing back to the root language === setting hasn't changed
+	// but if settings.locale_variant is not empty then we assume the user is trying to switch back to the root
```

Comment:
After re-reading this comment, I think it's not very clear, and actually not accurate. Sorry, my fault. Maybe this is better:

```
// if settings.locale_variant is empty we're dealing with an already-saved root language so we don't save the change
// conversely, if settings.locale_variant isn't empty we assume the user is trying to switch from the variant back to the root
```